ocaml/xapi/xapi_host.ml

```diff
@@ -66,6 +66,19 @@ let assert_safe_to_reenable ~__context ~self =
     List.iter (fun self -> Xapi_pif.abort_if_network_attached_to_protected_vms ~__context ~self) unplugged_pifs;
   end
 
+(* The maximum pool size allowed must be restricted to 3 hosts for the pool which does not have Pool_size feature *)
+let pool_size_is_restricted ~__context =
+  let cvm_exception =
+    let dom0 = Helpers.get_domain_zero ~__context in
+    Db.VM.get_records_where ~__context
+      ~expr:(Eq(Field "is_control_domain", Literal "true"))
+    |> List.exists (fun (vmref, vmrec) ->
+        (vmref <> dom0) &&
+        (Xapi_stdext_std.Xstringext.String.endswith "-CVM" vmrec.API.vM_name_label)
+      )
+  in
+  (not cvm_exception) && (not (Pool_features.is_enabled ~__context Features.Pool_size))
```

Comment:
So this means that we can have arbitrarily many VMs if we have a CVM but not the pool size feature?

Reply:
arbitrarily many hosts, yes.

Reply:
Oh ok - though this was not intentional, because I found this behaviour surprising.